components/browser/state/src/main/java/mozilla/components/browser/state/reducer/MediaSessionReducer.kt

```diff
@@ -0,0 +1,142 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.state.reducer
+
+import mozilla.components.browser.state.action.MediaSessionAction
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.state.SessionState
+import mozilla.components.browser.state.state.MediaSessionState
+import mozilla.components.concept.engine.mediasession.MediaSession
+
+internal object MediaSessionReducer {
+    /**
+     * [MediaSessionAction] Reducer function for modifying the [MediaSessionState] of a [SessionState].
+     */
+    fun reduce(state: BrowserState, action: MediaSessionAction): BrowserState {
+        return when (action) {
+            is MediaSessionAction.ActivatedMediaSessionAction ->
+                state.updateMediaSession(action.tabId, action.mediaSession)
+            is MediaSessionAction.DeactivatedMediaSessionAction ->
+                state.updateMediaSession(action.tabId, action.mediaSession)
+            is MediaSessionAction.UpdateMediaMetadataAction ->
+                state.updateMediaMetadata(action.tabId, action.mediaSession, action.metadata)
+            is MediaSessionAction.UpdateMediaPlaybackStateAction ->
+                state.updatePlaybackState(action.tabId, action.mediaSession, action.playbackState)
+            is MediaSessionAction.UpdateMediaFeatureAction ->
+                state.updateMediaFeature(action.tabId, action.mediaSession, action.features)
+            is MediaSessionAction.UpdateMediaPositionStateAction ->
+                state.updatePositionState(action.tabId, action.mediaSession, action.positionState)
+            is MediaSessionAction.UpdateMediaMutedAction ->
+                state.updateMuted(action.tabId, action.mediaSession, action.muted)
+            is MediaSessionAction.UpdateMediaFullscreenAction ->
+                state.updateFullscreen(action.tabId, action.mediaSession, action.fullScreen, action.elementMetadata)
+        }
+    }
+}
+
+private fun BrowserState.updateMediaSession(
+    tabId: String,
+    mediaSession: MediaSession
+): BrowserState {
+    return updateTabState(tabId) { current ->
+        current.createCopy(mediaSessionState = current.mediaSessionState.copy(
+            sessionId = tabId,
```

Comment:
It seems unnecessary to have the sessionId in the state object since it is already a child of the tab.

Reply:
That's true, this is redundant.  I'll remove.